include/ck/tensor_operation/gpu/device/device_elementwise_extension.hpp

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2022, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include <memory>
+#include <array>
+
+#include "ck/ck.hpp"
+#include "ck/utility/data_type.hpp"
+#include "ck/utility/tuple.hpp"
+#include "ck/utility/math_v2.hpp"
+
+#include "ck/tensor_operation/gpu/device/device_elementwise.hpp"
+
+namespace ck {
+namespace tensor_operation {
+namespace device {
+
+struct NormalizeInInfer
+{
+    NormalizeInInfer(double epsilon = 1e-4) : epsilon_(epsilon) {}
+
+    template <typename T1, typename T2, typename T3, typename T4>
+    __host__ __device__ constexpr void operator()(T1& y,
+                                                  const T1& x,
+                                                  const T2& mean,
+                                                  const T2& variance,
+                                                  const T3& gamma,
+                                                  const T4& beta) const
+    {
+        static_assert(std::is_same<T2, float>::value || std::is_same<T2, double>::value,
+                      "Data type is not supported by this operation!");
+
+        using ck::type_convert;
+        using ck::math::sqrt;
+
+        T2 tmp_x, tmp_y;
+
+        tmp_x = type_convert<T2>(x);
+
+        tmp_y = ((tmp_x - mean) / sqrt(variance + type_convert<T2>(epsilon_))) *
+                    type_convert<T2>(gamma) +
+                type_convert<T2>(beta);
+        y = type_convert<T1>(tmp_y);
+    };
+
+    double epsilon_;
+};
+
+template <typename XDataType,
+          typename YDataType,
+          typename ScaleDataType,
+          typename BiasDataType,
+          typename MeanVarDataType,
+          index_t Rank>
+using DeviceElementwiseForBatchNormInfer = ck::tensor_operation::device::DeviceElementwise<
```

Comment:
Is this class really necessary?

Reply:
I think this definition make the `GetInstances()` API easier to understand,  since the user does not have to use `Tuple` to construct the instance of  a template class of `DeviceElementwise`

Reply:
I think this is not necessary. If you can use the base class for doing the same thing, then the derived class is not needed.

If you think the user may not know how to use the baes class for BN-infer, you can write example for that.

Reply:
Solved by commit [dacd8](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/531/commits/31da74c1ac36558339455d578635a24b881dacd8)